test/e2e_env/multizone/meshservice/policies.go

```diff
@@ -225,4 +257,155 @@ spec:
 			g.Expect(response.Instance).To(HavePrefix("second-test-server"))
 		}, "30s", "1s").Should(Succeed())
 	})
+
+	It("should configure MeshRetry for MeshService in another zone", func() {
+		Expect(YamlK8s(fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshFaultInjection
+metadata:
+  name: mesh-fi-ms-another-zone
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+    kuma.io/origin: zone
+spec:
+  targetRef:
+    kind: Mesh
+    proxyTypes: ["Sidecar"]
+  from:
+    - targetRef:
+        kind: Mesh
+      default:
+          http:
+            - abort:
+                httpStatus: 503
+                percentage: "50"
+`, Config.KumaNamespace, meshName))(multizone.KubeZone2)).To(Succeed())
+		// given
+		// create a tunnel to test-client admin
+		Expect(multizone.KubeZone1.PortForwardService("test-client", namespace, 9901)).To(Succeed())
+		portFwd := multizone.KubeZone1.GetPortForward("test-client")
+		tnl := tunnel.NewK8sEnvoyAdminTunnel(multizone.Global.GetTesting(), portFwd.ApiServerEndpoint)
+
+		// then
+		Eventually(func(g Gomega) {
+			response, err := client.CollectFailure(
+				multizone.KubeZone1,
+				"test-client",
+				fmt.Sprintf("test-server.%s.svc.%s.mesh.local", namespace, Kuma2),
+				client.FromKubernetesPod(namespace, "test-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(response.ResponseCode).To(Equal(503))
+		}, "30s", "1s").Should(Succeed())
+
+		Eventually(func(g Gomega) {
+			g.Expect(retryStat(tnl)).To(stats.BeEqualZero())
+		}, "5s", "1s").Should(Succeed())
+
+		meshRetryPolicy := fmt.Sprintf(`
+apiVersion: kuma.io/v1alpha1
+kind: MeshRetry
+metadata:
+  name: mr-for-ms-in-zone-2
+  namespace: %s
+  labels:
+    kuma.io/mesh: %s
+    kuma.io/origin: zone
+spec:
+  targetRef:
+    kind: Mesh
+  to:
+    - targetRef:
+        kind: MeshService
+        labels:
+          kuma.io/display-name: test-server
+          kuma.io/zone: %s
+      default:
+        http:
+          numRetries: 6
+          retryOn: ["503"]
+`, Config.KumaNamespace, meshName, Kuma2)
+		// and when a MeshRetry policy is applied
+		Expect(framework.YamlK8s(meshRetryPolicy)(multizone.KubeZone1)).To(Succeed())
+
+		// then
+		Eventually(func(g Gomega) {
+			response, err := client.CollectEchoResponse(
+				multizone.KubeZone1,
+				"test-client",
+				fmt.Sprintf("test-server.%s.svc.%s.mesh.local", namespace, Kuma2),
+				client.FromKubernetesPod(namespace, "test-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(response.Instance).To(HavePrefix("kube-test-server-2"))
+		}, "30s", "1s").MustPassRepeatedly(3).Should(Succeed())
+
+		// and
+		Eventually(func(g Gomega) {
+			g.Expect(retryStat(tnl)).To(stats.BeGreaterThanZero())
+		}, "5s", "1s").Should(Succeed())
+
+		// remove Policies
+		Expect(DeleteMeshPolicyOrError(multizone.KubeZone1, meshretry_api.MeshRetryResourceTypeDescriptor, "mr-for-ms-in-zone-2")).To(Succeed())
```

Comment:
you should probably add MeshRetryResourceTypeDescriptor to `E2EAfterEach` section we we delete policies between tests